core/quality/morqa_transport.go

```diff
@@ -58,11 +58,30 @@ func mapEventToMetric(event Event) (string, *metrics.Event) {
 		return proposalEventToMetricsEvent(event.Context.(market.ServiceProposal), event.Application)
 	case traceEventName:
 		return traceEventToMetricsEvent(event.Context.(sessionTraceContext), event.Application)
+	case registerIdentity:
+		return identityRegistrationEvent(event.Context.(registrationEvent), event.Application)
 	}
 
 	return "", nil
 }
 
+func identityRegistrationEvent(data registrationEvent, info appInfo) (string, *metrics.Event) {
+	return data.Identity, &metrics.Event{
+		Metric: &metrics.Event_RegistrationPayload{
+			RegistrationPayload: &metrics.RegistrationPayload{
+				Version: &metrics.VersionPayload{
+					Version: info.Version,
+					Os:      info.OS,
+					Arch:    info.Arch,
+				},
+				Country:  data.Country,
+				Identity: data.Identity,
```

Comment:
All metrics are originated with identity, no need to repeat this

Reply:
From morqa code I assumed that this is something that might get changed or is not ideal.
I think we should keep it here, but we can argue about this 🤷

Reply:
No, it's our agreed design, everyone  metric has `event.TargetId` 
Furthermore, all metrics are signed with identity PK, and it prevents from forging metrics.